manifests/init.pp

```diff
@@ -33,156 +33,129 @@
 class jira (
 
   # Jira Settings
-  $version      = '6.4.1',
-  $product      = 'jira',
-  $format       = 'tar.gz',
-  $installdir   = '/opt/jira',
-  $homedir      = '/home/jira',
-  $user         = 'jira',
-  $group        = 'jira',
-  $uid          = undef,
-  $gid          = undef,
-  $shell        = '/bin/true',
-
+  String $version                                                   = '6.4.1',
+  String $product                                                   = 'jira',
+  String $format                                                    = 'tar.gz',
+  Stdlib::Absolutepath $installdir                                  = '/opt/jira',
+  Stdlib::Absolutepath $homedir                                     = '/home/jira',
+  String $user                                                      = 'jira',
+  String $group                                                     = 'jira',
+  $uid                                                              = undef,
+  $gid                                                              = undef,
+  Stdlib::Absolutepath $shell                                       = '/bin/true',
```

Comment:
Whilst googling, I came across this style too. https://github.com/jlambert121/jlambert121-puppet/blob/master/manifests/init.pp